src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/__tests__/HealthCertificate+BelongsToSamePersonTests.swift

```diff
@@ -0,0 +1,324 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import XCTest
+@testable import ENA
+import HealthCertificateToolkit
+
+class HealthCertificate_BelongsToSamePersonTests: XCTestCase {
+
+	func test_BelongsToSamePerson() throws {
+		for testCase in testCases {
+			let lhsCertificate = try HealthCertificate(
+				base45: try base45Fake(
+					from: DigitalCovidCertificate.fake(
+						name: Name.fake(
+							standardizedFamilyName: testCase.actHolderA.fnt,
+							standardizedGivenName: testCase.actHolderA.gnt
+						),
+						dateOfBirth: testCase.actHolderA.dob
+					)
+				),
+				validityState: .valid
+			)
+			
+			let rhsCertificate = try HealthCertificate(
+				base45: try base45Fake(
+					from: DigitalCovidCertificate.fake(
+						name: Name.fake(
+							standardizedFamilyName: testCase.actHolderB.fnt,
+							standardizedGivenName: testCase.actHolderB.gnt
+						),
+						dateOfBirth: testCase.actHolderB.dob
+					)
+				),
+				validityState: .valid
+			)
+			
+			XCTAssertEqual(lhsCertificate.belongsToSamePerson(rhsCertificate), testCase.expIsSameHolder, "Test failed. Description: \(testCase.description)")
+		}
+	}
+	
+	struct TestCase {
+		struct Holder {
+			let gnt: String
+			let fnt: String
+			let dob: String
+		}
+		
+		let description: String
+		let actHolderA: Holder
+		let actHolderB: Holder
+		let expIsSameHolder: Bool
+	}
+	
+	let testCases = [
+		TestCase(
```

Comment:
Can you dynamically read this from JSON?

Reply:
@mlenkeit  There was only a yaml, when I was writing the tests.

Reply:
Added dynamic test case loading.
Discovered unimplemented edge cases.
I like this test driven thingy ;)